compiler/rustc_session/src/options.rs

```diff
@@ -1810,6 +1810,8 @@ options! {
         "use a more precise version of drop elaboration for matches on enums (default: yes). \
         This results in better codegen, but has caused miscompilations on some tier 2 platforms. \
         See #77382 and #74551."),
+    prefer_deps_of_dynamic: bool = (false, parse_bool, [TRACKED],
+        "prefer linking to static dependencies of dynamic libraries over available dynamic libraries (default: no)"),
```

Comment:
If it works robustly, I feel like this should be unconditionally enabled.

Reply:
I agree, but I'm not confident in there being no regressions. We don't really have many dylib and multi crate type tests.